python-cffi/netplan/netdef.py

```diff
@@ -298,6 +298,23 @@ def to_dict(self):
 
         return route
 
+    def __hash__(self):
+        return hash(
+            (self.to, self.via,
+             self.from_addr, self.table,
+             self.family, self.type, self.scope))
+
+    def __eq__(self, route):
```

Comment:
I decided to remove the `metric` from comparisons because of some differences between NM and networkd. For example, if you don't defined a metric in the YAML, NM will install the route with metric 100 and networkd will not set a metric. In this case the route read from the YAML wouldn't match the system's. On VRFs, if the metric is not defined, NM will set it to 470. So, it was making it harder to compare routes and I tried to reduce the noise.

Reply:
I think it's still an important information to have and we should consider putting it back.